### PR TITLE
Add rpmbuild support for Travis CI

### PR DESCRIPTION
**Pull request by brianhlin**
I'm not 100% sure if the example works but I can update it after I play around with this in a real Travis environment

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,39 @@
 FROM centos:7
 
+LABEL maintainer OSG Software <help@opensciencegrid.org>
```
Comment (matyasselmeci) [inline Dockerfile]:
```suggestion
LABEL maintainer="OSG Software <help@opensciencegrid.org>"
```

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,39 @@
 FROM centos:7
 
+LABEL maintainer="OSG Software <help@opensciencegrid.org>"
 LABEL name="OSG 3.4 OSG-Build client"
 
 RUN yum -y install https://repo.opensciencegrid.org/osg/3.4/osg-3.4-el7-release-latest.rpm && \
     yum -y install epel-release \
                    yum-plugin-priorities && \
+    # Install packages included in the Koji build repos
+    yum -y install epel-rpm-macros \
```
Comment (matyasselmeci) [inline Dockerfile]:
```suggestion
    yum -y install --enablerepo=osg-development epel-rpm-macros \
```

Comment (matyasselmeci) [reply Dockerfile]: You're installing buildsys-macros twice. You should probably just install everything with osg-development enabled (once the repo is available). Possibly use minefield.

Comment (matyasselmeci) [reply Dockerfile]: er, I was thinking combine this with the yum install command below.

File: build-from-github [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Generate the dirs/files necessary for osg-build
+mkdir upstream
+VERSION=$(awk '/Version:/ {print $2}' "/$REPO_NAME/rpm/$REPO_NAME.spec")
```
Comment (matyasselmeci) [inline build-from-github]:
```suggestion
VERSION=$(awk '/Version:/ {print $2}' "/$REPO_NAME/rpm/$REPO_NAME.spec" | head -n 1)
```
Subpackages may have their own versions so you need to take the top one.
This is pretty brittle in general - if something takes its version based on a macro expansion for example, this wouldn't work. Googling hasn't found me a better way of doing this so I suggest adding a command-line argument to override either the version or the tarball name.

Comment (brianhlin) [reply build-from-github]: How about using this to handle macros? https://stackoverflow.com/questions/3634650/can-i-use-rpm-to-expand-the-macros-in-a-specfile/7901349#7901349

Comment (matyasselmeci) [reply build-from-github]: Goddammit, why did you find that and I didn't?!

Yeah sure, give it a shot.

Comment (brianhlin) [reply build-from-github]: Yup, it works. The only annoying thing is that it's not available in EL6 so we'll have to find something else if we want to expand local build capability to EL6

Comment (matyasselmeci) [reply build-from-github]: OK well then put the manual override in anyway.

**Comment by brianhlin** [comment]
Also, the example should work. I just tested param expansion in an env var specified in `.travis.yml`.

**Pull request merged**